lib/_stream_readable.js

```diff
@@ -515,9 +518,26 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
       ondrain();
   }
 
+  function maybeCork() {
+    if (!autoCork || corked) {
+      debug('already corked');
+      return;
+    }
+    debug('corking');
+    corked = true;
+    dest.cork();
+    process.nextTick(function () {
```

Comment:
I wonder if pulling this anonymous function out of `maybeCork()` would help performance-wise?